storage/src/backend/connection.rs

```diff
@@ -428,90 +452,88 @@ impl Connection {
+                    }
+                    Err(err) => {
                         warn!(
-                            "reach to fail limit {}, disable mirror: {}",
-                            mirror.failure_limit,
-                            format!("{:?}", mirror).to_lowercase()
+                            "request mirror server failed, mirror: {:?},  error: {:?}",
+                            mirror, err
                         );
-                        mirror.status.store(false, Ordering::Relaxed);
+                        mirror.failed_times.fetch_add(1, Ordering::Relaxed);
+
+                        if mirror.failed_times.load(Ordering::Relaxed) >= mirror.failure_limit {
+                            warn!(
+                                "reach to failure limit {}, disable mirror: {:?}",
+                                mirror.failure_limit, mirror
+                            );
+                            mirror.status.store(false, Ordering::Relaxed);
+
+                            let mut idx = 0;
+                            loop {
+                                if idx == self.mirror_state.mirrors.len() {
+                                    break None;
+                                }
+                                let m = &self.mirror_state.mirrors[idx];
+                                if m.status.load(Ordering::Relaxed) {
+                                    warn!("mirror server has been changed to {:?}", m);
+                                    break Some(m);
+                                }
 
-                        let mut idx = 0;
-                        loop {
-                            if idx == self.mirror_state.mirrors.len() {
-                                break None;
-                            }
-                            let m = &self.mirror_state.mirrors[idx];
-                            if m.status.load(Ordering::Relaxed) {
-                                warn!(
-                                    "mirror server has been changed to {}",
-                                    format!("{:?}", m).to_lowercase()
-                                );
-                                break Some(m);
+                                idx += 1;
                             }
-
-                            idx += 1;
+                            .map(|m| self.mirror_state.current.store(Some(m.clone())))
+                            .unwrap_or_else(|| self.mirror_state.current.store(None));
                         }
-                        .map(|m| self.mirror_state.current.store(Some(m.clone())))
-                        .unwrap_or_else(|| self.mirror_state.current.store(None));
                     }
                 }
+                // FIXME: Move to the next available mirror directly when mirror health checking is available.
+                warn!("Failed to request mirror server, fallback to original server.");
+                for (key, _) in mirror.config.headers.iter() {
```

Comment:
Why remove all entries here?

Reply:
Maybe give some comments and use `mirror.config.headers.clear()`.

Reply:
> Why remove all entries here?

Just remove mirror-related headers, avoid sending them to the original registry.